ipatests/test_integration/test_installation.py

```diff
@@ -1075,3 +1075,60 @@ def test_replica_install_against_server(self):
                                             self.replicas[0].hostname],
                                            stdin_text=dirman_password)
         assert self.replicas[0].hostname not in cmd.stdout_text
+
+
+class TestIpaJoin(IntegrationTest):
+
+    num_clients = 2
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master, setup_dns=False)
+
+    def test_unenroll_option(self):
+        """
+        Test ipa-join Using unenroll option
+        """
+        # install client and kinit admin
+        tasks.install_client(self.master, self.clients[0])
+        tasks.kinit_admin(self.clients[0])
+
+        # checking unenroll option
+        result = self.clients[0].run_command(['ipa-join', '-u',
+                                             self.clients[0].hostname])
+        assert "Unenrollment successful." in result.stderr_text
```

Comment:
This verifies that the command returns success, not that it did what it said it did.

ipa-join -u does a kinit using the host principal then calls the ipa host-disable <fqdn> command. You could, for example, do a host-show to ensure that it has no kerberos keys. To really stretch it any services on the host should also be disabled.

It is probably worth calling join -u twice, expecting success the first and failure the second..